File: c/stack.c
//program to perform stack opertions .
#include<stdio.h>
#include<stdlib.h>
#define n 5
int top=-1,i,stack[n];
void push()
{
    int ele;
    if(top==n-1)
    {
        printf("Stack is full\n");
    }
    else
    {
        printf("Enter the element \n");
        scanf("%d",&ele);
        stack[++top]=ele;
    }

}
void pop()
{
    if(top==-1)
    {
        printf("stack is empty\n");
    }
    else
    {
        printf("poped element is %d\n",stack[top--]);
    }
}
void display()
{
    if(top==-1)
    {
        printf("stack is empty\n");
    }
    printf("Elements of stack  are \n");
    
    for(i=0;i<=top;i++)
    {
        printf("%d ",stack[i]);
    }
}
void main()
{
    int ch;
    while(1)
    {
        printf("\n__stack__operation__\n");
        printf("1.push\n2.pop\n3.display\n4.exit()\n");
        printf("Enter your choice\n");
        scanf("%d",&ch);
        switch(ch)
        {
            case 1:push();
                   break;
            case 2:pop();
                   break;
            case 3:display();
                   break;
            case 4:exit(0);
        }
    }
}

File: java/stack.java
import java.util.*;

class implement {

    private int top;
    private int stk[];

    implement(int size) {

        top = -1;

        stk = new int[size];
    }

    void push(int data) {
        if (top == stk.length - 1) {
            System.out.println("Stack is full");
        } else {

            stk[++top] = data;

            System.out.println("Element pushed is :" + stk[top]);
        }
    }

    int pop() {
        if (top == -1) {
            System.out.println("Stack is empty");
        } else {
            System.out.println("Element poped is ::" + stk[top]);
        }
        return stk[top--];
    }

    void display() {
        if (top < 0) {
            System.out.println("Stack is empty:");
        } else {
            System.out.println("The contents of Stack is ::");
            for (int i = top; i >= 0; i--) {
                System.out.println(stk[i] + " ");
            }
        }
    }

}

public class stack {

    public static void main(String args[]) {
        int ch, size;

        Scanner in = new Scanner(System.in);

        System.out.println("Enter the size of stack");
        size = in.nextInt();
        boolean x = true;
        implement i = new implement(size);

        do {
            System.out.println("---Stack operation in java---");
            System.out.println("1)Push\t2)Pop\t3)Display\t4)Exit");

            ch = in.nextInt();

            switch (ch) {
                case 1:
                    System.out.println("Enter the Element ");
                    i.push(in.nextInt());
                    break;
                case 2:
                    i.pop();
                    break;
                case 3:
                    i.display();
                    break;
                case 4:
                    x = false;
                    System.out.println("exit ");
            }

        } while (x == true);

    }

}
